client/blocks/image-selector/preview.jsx

```diff
@@ -0,0 +1,223 @@
+					return e;
+				} )
+		);
+
+		this.setState( { images }, () => {
+			if ( 'function' === typeof callback ) {
+				callback();
+			}
+		} );
+		defer( () => {
+			const imageIds = images.map( image => image.ID );
+			if (
+				onImageChange &&
+				images &&
+				images.length === itemIds.length &&
+				! isEqual( imageIds, itemIds )
+			) {
+				onImageChange( images );
+			}
+		} );
+	};
+
+	src = image => {
+		return url( image, {
+			maxWidth: this.props.maxWidth,
+			size: 'post-thumbnail',
+		} );
+	};
+
+	renderPlaceholder = image => {
+		const { placeholder } = image;
+		return (
+			<figure>
+				<img src={ placeholder || '' } alt="" />
+			</figure>
+		);
+	};
+
+	renderUploaded = ( { URL, placeholder } ) => {
+		return (
+			<figure>
+				<ImagePreloader
+					src={ URL }
+					placeholder={ placeholder ? <img src={ placeholder } alt="" /> : <span /> }
+				/>
+			</figure>
+		);
+	};
+
+	renderImage = image => {
+		const src = this.src( image );
+		const { onImageClick, onRemoveImage, showEditIcon, translate } = this.props;
+		const id = image.ID || image.transientId;
+
+		const removeImage = () => {
+			onRemoveImage( image );
+		};
+
+		const classes = classNames( 'image-selector__item', {
+			'is-transient': image.transient || ! src,
+			preview: null === src,
+		} );
+
+		return (
+			<div className={ classes } key={ id }>
+				<Button
+					className="image-selector__image"
+					onClick={ onImageClick }
+					borderless
+					compact
+					data-tip-target="image-selector-image"
+				>
+					<Spinner />
+					{ src ? this.renderUploaded( image ) : this.renderPlaceholder( image ) }
+					{ showEditIcon && <Gridicon icon="pencil" className="image-selector__edit-icon" /> }
+				</Button>
+				<Button
+					onClick={ removeImage }
+					compact
+					aria-label={ translate( 'Remove image' ) }
+					className="image-selector__remove"
+				>
+					<Gridicon icon="cross-small" size={ 24 } className="image-selector__remove-icon" />
+				</Button>
+			</div>
+		);
+	};
+
+	renderUploadPlaceholder() {
+		const { onImageClick, translate, compact, multiple } = this.props;
+
+		const classes = classNames( 'image-selector__uploader-wrapper', {
+			compact,
+		} );
+
+		const addString = multiple ? translate( 'Add images' ) : translate( 'Add image' );
+		const iconSize = compact ? 24 : 36;
+
+		return (
+			<Button
+				className={ classes }
+				onClick={ onImageClick }
+				borderless
+				compact
+				data-tip-target="image-selector-image"
+			>
+				<div className="image-selector__uploader-picker">
+					<div className="image-selector__uploader-label">
+						<div>
+							<Gridicon icon="add-image" size={ iconSize } />
+							<p>{ ! compact && addString }</p>
+						</div>
+					</div>
+				</div>
+			</Button>
+		);
+	}
+
+	render() {
+		const images = [ ...this.state.images ];
+		const featuredImage = ( images && images.shift() ) || null;
+
+		return (
+			<div className="image-selector__images-wrapper">
```

Comment:
Do you think having a `classname` prop would be of use here so when this is used in other components the styling could be adjusted accordingly?